system-tests/src/test/java/org/eclipse/edc/samples/transfer/Transfer05fileTransferCloudTest.java

```diff
@@ -0,0 +1,234 @@
+                    Map.entry("web.http.path", "/api"),
+                    Map.entry("web.http.management.port", "19193"),
+                    Map.entry("web.http.management.path", "/management"),
+                    Map.entry("web.http.protocol.port", "19194"),
+                    Map.entry("web.http.protocol.path", "/protocol"),
+                    Map.entry("edc.api.auth.key", "password"),
+                    Map.entry("edc.transfer.proxy.token.signer.privatekey.alias", "private-key"),
+                    Map.entry("edc.transfer.proxy.token.verifier.publickey.alias", "public-key"),
+                    Map.entry("web.http.public.port", "19291"),
+                    Map.entry("web.http.public.path", "/public"),
+                    Map.entry("web.http.control.port", "19192"),
+                    Map.entry("web.http.control.path", "/control"),
+                    Map.entry("edc.dataplane.api.public.baseurl", "http://localhost:19291/public"),
+                    Map.entry("edc.converter.usefilesystem", "false"),
+                    Map.entry("edc.vault.hashicorp.url", "http://127.0.0.1:8200"),
+                    Map.entry("edc.vault.hashicorp.token", "<root-token>"),
+                    Map.entry("edc.vault.hashicorp.api.secret.path", "/v1/secret"),
+                    Map.entry("edc.vault.hashicorp.health.check.enabled", "false"),
+                    Map.entry("edc.blobstore.endpoint.template", "http://127.0.0.1:" + getAzuritePort() + "/%s"),
+                    Map.entry("edc.aws.access.key", "accessKeyId"),
+                    Map.entry("edc.aws.secret.access.key", "secretAccessKey")
+            ),
+            PROVIDER_MODULE_PATH
+    ));
+
+    @Test
+    void pushFile() throws Exception {
+
+        MinioClient minioClient =
+                MinioClient.builder()
+                .endpoint("http://" + minioContainer.getHost() + ":" + minioContainer.getMappedPort(MINIO_PORT))
+                .credentials(MINIO_ACCOUNT_NAME, MINIO_ACCOUNT_KEY)
+                .build();
+
+        minioClient.makeBucket(MakeBucketArgs.builder().bucket(MINIO_BUCKET_NAME).build());
+
+        var requestBody = getFileContentFromRelativePath(START_TRANSFER_FILE_PATH)
+                .replace("http://localhost:9000", "http://localhost:" + minioContainer.getMappedPort(9000).toString());
+
+        var contractAgreementId = runNegotiation();
+        
+        var transferProcessId = startTransfer(requestBody, contractAgreementId);
+
+        checkTransferStatus(transferProcessId, TransferProcessStates.COMPLETED);
+
+
+        Iterable<Result<Item>> results = minioClient.listObjects(
+                ListObjectsArgs.builder().bucket(MINIO_BUCKET_NAME).build());
+
+        Result<Item> result = results.iterator().next();
+        Item item = result.get();
+
+        boolean objectFound = item.objectName().equals(FILE_NAME);
+
+        assertTrue(objectFound, "test-document.txt could not be found in minio");

```

Comment:
please use `assertJ` assertions instead of the base junit ones.
these lines could become something like:
```java
        assertThat(objects)
                .isNotEmpty().first()
                .extracting(result -> {
                    try {
                        return result.get();
                    } catch (Exception e) {
                        return fail();
                    }
                })
                .satisfies(item -> assertThat(item.objectName()).isEqualTo(FILE_NAME));
```